hibernate-reactive-routes-quickstart/src/main/java/org/acme/hibernate/reactive/FruitsRoutes.java

```diff
@@ -37,55 +37,45 @@
-    public Uni<Fruit> getSingle(@Param String id) {
-        return session.find(Fruit.class, Integer.valueOf(id));
+    public Uni<Fruit> getSingle(@Param Integer id) {
+        return session.find(Fruit.class, id);
     }
 
     @Route(methods = POST, path = "/")
     public Uni<Fruit> create(@Body Fruit fruit, HttpServerResponse response) {
         if (fruit == null || fruit.getId() != null) {
             return Uni.createFrom().failure(new IllegalArgumentException("Fruit id invalidly set on request."));
         }
-        return session.persist(fruit)
-                .chain(session::flush)
-                .onItem().transform(ignore -> {
-                    response.setStatusCode(201);
-                    return fruit;
-                });
+        return session.withTransaction(tx -> session.persist(fruit))
+                .invoke(() -> response.setStatusCode(201))
+                .replaceWith(fruit);
     }
 
     @Route(methods = PUT, path = "/:id")
-    public Uni<Fruit> update(@Body Fruit fruit, @Param String id) {
+    public Uni<Fruit> update(@Body Fruit fruit, @Param Integer id) {
         if (fruit == null || fruit.getName() == null) {
             return Uni.createFrom().failure(new IllegalArgumentException("Fruit name was not set on request."));
         }
-        return session.find(Fruit.class, Integer.valueOf(id))
-                // If entity exists then update
-                .onItem().ifNotNull().transformToUni(entity -> {
-                    entity.setName(fruit.getName());
-                    return session.flush()
-                            .onItem().transform(ignore -> entity);
-                })
-                // else
+        return session.withTransaction(tx -> session.find(Fruit.class, id)
+                // If entity exists then update it
+                .onItem().ifNotNull().invoke(entity -> entity.setName(fruit.getName())))
+                // If entity not found, fail
                 .onItem().ifNull().fail();
     }
 
     @Route(methods = DELETE, path = "/:id")
-    public Uni<Fruit> delete(@Param String id, HttpServerResponse response) {
-        return session.find(Fruit.class, Integer.valueOf(id))
-                // If entity exists then delete
-                .onItem().ifNotNull().transformToUni(entity -> session.remove(entity)
-                        .chain(session::flush)
-                        .map(ignore -> {
-                            response.setStatusCode(204).end();
-                            return entity;
-                        }))
-                // else
+    public Uni<Fruit> delete(@Param Integer id, HttpServerResponse response) {
+        return session.withTransaction(tx -> session.find(Fruit.class, id)
+                // If entity exists then delete it
+                .onItem().ifNotNull()
+                    .call(entity -> session.remove(entity)
+                            .invoke(() -> response.setStatusCode(204))))
```

Comment:
Same incredibly minor detail here.

Reply:
If I do that it will set the status code even when the entity doesn't exist